dht_options.go

```diff
@@ -91,6 +91,10 @@ func (c *config) applyFallbacks(h host.Host) error {
 			return fmt.Errorf("the default validator was changed without being marked as changed")
 		}
 	}
+
+	if !c.enableValues {
+		c.validator = nil
+	}
```

Comment:
@Stebalien do you think this is worth doing or no? We have some tests and I've done a once-over and it looks like we shouldn't have any problems with `nil` validators if `enableValues=false`, if it's `enableValues=true` though is it fair to say the developer has shot themselves in the foot or should we throw in a bunch of `nil` checks or implement a `NilValidator` that's applied here in the fallbacks.

Reply:
Why not just leave the validator?

Reply:
I guess no particular reason, it's not like it's going to eat up much memory. This was more a point for discussing what we should do around `nil` validators.

I'll remove those few lines and merge.